README.md

```diff
@@ -1,2 +1,106 @@
+
+### Set up an IDE (optional)
+
+If for some reason, you need/want an IDE deployed (if for example you are working on a server). Install via Helm [a vscode service](https://artifacthub.io/packages/helm/inseefrlab/vscode) then access by port-forwarding `kubectl -n argo port-forward deployment/my-vscode 35547:8080`. You will probably have to give admin rights to this VSCode (service account: `my-vscode` over MinIO (as done for Argo)). Finally, you will also have to add Metaflow configuration variables to the ConfigMap of VSCode, for putting metaflow code packages in MinIO. 
+
+![configmap-my-vscode-configmaps3metaflow](https://github.com/SDSC-ORD/zarr-linked-data/assets/22447169/498a1306-5289-4527-9cfa-22e947d555e3)
+
+Then you just install the requirements with `pip install -r requirements.txt` and you're set to go! 
+
+# C. Example/Usage
+
+## Fake Zarr Data
+You may not have data under the zarr format yet. You can define some metadata instance objects (such as in `zarr_linked_data/data/original_data.jsonld`), then define your hierarchy levels, data level and other parameters for the creation of a random Zarr test store using `zarr_linked_data/fake_data_flow.py`.
+You can find all our example data in the `data` folder, from the original jsonld metadata instance data, to the `test_store.zarr` containing random arrays as well as the jsonld metadata.
+
+## Local scripts
+
+- Explore Zarr group and attributes functionalities with `python zarr_linked_data/local_dev/zarr_explore.py`
+- Fake data creation pipeline with: `python zarr_linked_data/fake_data_flow.py`
+- Metadata consolidation Metaflow pipeline with: `python zarr_linked_data/local_dev/consolidate_metadata_flow.py run`
+- Retrieve URI Metaflow pipeline with: `python zarr_linked_data/local_dev/retrieval_flow.py run`
+
+## Scripts on kubernetes
+
+- Fake data creation pipeline with: `python zarr_linked_data/fake_data_flow.py` (same as for locally)
+- Metadata consolidation Metaflow pipeline with: `python zarr_linked_data/consolidate_metadata_flow.py run`
+- Retrieve URI Metaflow pipeline with: `python zarr_linked_data/retrieval_flow.py run`
+
+You can check the correct run of the metaflow flows with the command specified in the script. Then you will need to send them to Argo workflows. [Using Metaflow to create Argo DAGs](https://docs.metaflow.org/production/scheduling-metaflow-flows/scheduling-with-argo-workflows): `python zarr_linked_data/retrieval_flow.py --with retry argo-workflows create` (same for `consolidate_metadata_flow`)
+
+# Roadmap
+
+- Retrieval flow will be converted to a FastAPI instead
+- A flow `metadata update` will be added: it will transform the consolidated metadata and add it to a graph database (such as GraphDB or ApacheJenaFuseki): local development of [this flow is on an annex branch here](https://github.com/SDSC-ORD/zarr-linked-data/blob/metadata_extractor/zarr_linked_data/extract_all_metadata.py)
```

Comment:
**note:** awesome! I'm very interested :grinning:

Reply:
If I can, I will make it for local_dev in my refactor next week

Reply:
Actually I will first do a PR for local dev and another one for this extra flow